File: packaging/uniformBuild/scripts/actions.py
import requests

OWNER ="aws"
REPO = "amazon-cloudwatch-agent"
GITHUB_API = f"https://api.github.com/repos/{OWNER}/{REPO}/actions"
id = "6202732096"
GITHUB_API_RUN = GITHUB_API + f"/runs/{id}"
GITHUB_API_JOB =  GITHUB_API_RUN + "/jobs?per_page=100"
data = requests.get(GITHUB_API_RUN).json()
print(data)
print(data['status'],data['conclusion'])
if data.get('conclusion') =='failure':
    jobs = []
    for page in range(3):
        jobdata = requests.get(GITHUB_API_JOB+f"&page={page}").json()
        jobs += jobdata.get('jobs')
    print(len(jobs))
    failed = []
    job_ids = []
    for job in jobs:
        id = job.get('id')
        if id in job_ids:
            continue
        # print(job.get("name"),job.get("conclusion"))
        if job.get("conclusion") == 'failure':
            failed.append(job.get("name"))
            # print(job.get("name"),"failed")
        job_ids.append(id)
    print(len(failed),"\n".join(failed))

File: packaging/uniformBuild/scripts/actions.go
package main

import (
	"context"
	"fmt"
	"github.com/google/go-github/v55/github"
)

func main() {
	// Create a new GitHub client
	client := github.NewClient(nil)

	// Get information about a workflow run
	owner := "aws"
	repo := "amazon-cloudwatch-agent"
	var runID int64 = 6202732096
	run, _, err := client.Actions.GetWorkflowRunByID(context.Background(), owner, repo, runID)
	//run, _, err:= client.Actions.
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	jobs, _, err := client.Actions.ListWorkflowJobs(context.TODO(), owner, repo, runID, &github.ListWorkflowJobsOptions{
		Filter: "latest",
		ListOptions: github.ListOptions{
			Page:    1,
			PerPage: 300,
		},
	})
	fmt.Println("ID:", run.GetID())
	fmt.Println("Status:", run.GetStatus())
	fmt.Println("Conclusion:", run.GetConclusion())
	fmt.Println("Created At:", run.GetCreatedAt())
	fmt.Println("jobs", *jobs.TotalCount)
}
